app/assets/javascripts/application.js.erb

```diff
@@ -727,4 +727,19 @@ onPageLoad(() => {
         response.text().then(text => replace(comment, text));
       });
   });
+
+  on('click', '.comment_unread', (event) => {
+    var nodes = document.getElementsByClassName('comment_unread')
+    var foundIndex = Array.from(nodes).findIndex(function(n){
```

Comment:
Does this need to add an `id` to the comments? It feels like this could be `.findIndex((n) => { n == event.target })`.

Reply:
You were correct, the id wasn't needed after all.

I've also cleaned up the logic and added the "wrap around" feature that you requested. So now, when you click on the final comment it scrolls to the first one on the page.